extensions/podman/src/podman-configuration.ts

```diff
@@ -93,6 +95,82 @@ export class PodmanConfiguration {
     ) {
       await extensionApi.proxy.setProxy(proxySettings);
     }
+
+    // If we are on Mac, we need to monitor the configuration file to handle Rosetta changes
+    if (isMac()) {
+      extensionApi.configuration.onDidChangeConfiguration(async e => {
+        if (e.affectsConfiguration(`podman.${configurationRosetta}`)) {
+          await this.handleRosettaSetting();
+        }
+      });
+    }
+  }
+
+  async handleRosettaSetting(): Promise<void> {
+    // If the configuration does not exist, we will default to true
+    // if true, when we do updateRosettaSetting, if there is no configuration file, it will do nothing.
+    const useRosetta = extensionApi.configuration.getConfiguration('podman').get<boolean>(configurationRosetta) ?? true;
+    await this.updateRosettaSetting(useRosetta);
+  }
+
+  async updateRosettaSetting(useRosetta: boolean): Promise<void> {
+    // Initalize an empty configuration file for us to use
+    const containersConfContent = {
+      containers: toml.Section({}),
+      engine: toml.Section({
+        env: [] as string[],
+      }),
+      machine: toml.Section({}),
+      network: toml.Section({}),
+      secrets: toml.Section({}),
+      configmaps: toml.Section({}),
+    };
```

Comment:
Why are you creating an intermediary structure for the config? 

Isn't it possible to work directly on the result of `toml.parse(containersConfigFile)`, so even if the config file evolves and there are some new sections, it will still work?

Reply:
unable too as the information passed back by `toml.parse` is any and typescript complains.

I tried implementing it similar to your suggestion, but wasn't able to find a good solution. The code below (proxy update) by @benoitf was the inspiration I took this from.

this initializing + making sure that the information gets filled by checking the saved information would ensure that it all gets passed through.